service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchAwaiter.kt

```diff
@@ -205,6 +209,16 @@ class BatchAwaiter(
     }
   }
 
+  fun saveProgress(dbRunPartition: DbRunPartition) {
+    dbRunPartition.pkey_cursor = pkeyCursor
```

Comment:
I guess this doesn't need to be synchronized because there is still only a single underlying thread in backfila?

Reply:
per lease, yea, otherwise a lot of this code has to change

Reply:
yeah this should be fine.